pkg/policy/mapstate_test.go

```diff
@@ -96,162 +92,146 @@ func (ms *mapState) validatePortProto(t *testing.T) {
 }
 
 func (e mapStateEntry) withProxyPort(proxyPort uint16) mapStateEntry {
-	if proxyPort == 0 {
-		e.priority = 0
-	} else if e.priority == 0 {
-		e.priority = proxyPort // default for tie-breaking
-	}
-	e.ProxyPort = proxyPort
+	e.MapStateEntry = e.MapStateEntry.WithProxyPort(proxyPort)
 	return e
 }
 
-func (e mapStateEntry) withProxyPortPriority(proxyPort, priority uint16) mapStateEntry {
-	if proxyPort == 0 {
-		priority = 0
-	} else if priority == 0 {
-		priority = proxyPort // default for tie-breaking
-	}
-	e.ProxyPort = proxyPort
-	e.priority = priority
+func (e mapStateEntry) withProxyPortPriority(proxyPort uint16, priority uint8) mapStateEntry {
+	e.MapStateEntry = e.MapStateEntry.WithProxyPort(proxyPort).WithProxyPriority(priority)
 	return e
 }
 
 func TestMapState_denyPreferredInsertWithChanges(t *testing.T) {
```

Comment:
This test should probably be renamed now that `(*mapState).denyPreferredInsertWithChanges` was renamed:

```suggestion
func TestMapState_insertWithChanges(t *testing.T) {
```

Reply:
Done!